pkg/api/types.go

```diff
@@ -1456,3 +1456,94 @@ func AddToNodeAddresses(addresses *[]NodeAddress, addAddresses ...NodeAddress) {
 		}
 	}
 }
+
+// AutoScaler monitors other resources that are resizeable and adjusts them according to configuration.
+type AutoScaler struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the auto-scaler targets and thresholds.
+	Spec AutoScalerSpec `json:"spec,omitempty"`
+
+	// Status defines the actions the auto-scaler has taken.
+	Status AutoScalerStatus `json:"status,omitempty"`
+}
+
+// AutoScalerStatus provides the status of an auto-scaler.
+type AutoScalerStatus struct {
+	// TODO: open for discussion on what meaningful information can be reported in the status
+	// The status may return the replica count here but we may want more information
+	// such as if the count reflects a threshold being passed.
+}
+
+// AutoScalerSpec defines the auto-scaler targets and thresholds.
+type AutoScalerSpec struct {
+	// Thresholds holds a collection of AutoScaleThresholds that drive the auto-scaler.
+	Thresholds []AutoScaleThreshold `json:"thresholds,omitempty"`
+
+	// Enabled turns auto scaling on or off.
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
I can see merit in enabled for RC and autoscaler from the naive user perspective (what, you want me to delete this?!?) if you need a temporary reprieve from the behavior.  Agree however it's a big shift.

> On Mar 16, 2015, at 3:05 AM, David Oppenheimer <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > +}
> > +
> > +// AutoScalerStatus provides the status of an auto-scaler.
> > +type AutoScalerStatus struct {
> > +	// TODO: open for discussion on what meaningful information can be reported in the status
> > +	// The status may return the replica count here but we may want more information
> > +	// such as if the count reflects a threshold being passed.
> > +}
> > +
> > +// AutoScalerSpec defines the auto-scaler targets and thresholds.
> > +type AutoScalerSpec struct {
> > +	// Thresholds holds a collection of AutoScaleThresholds that drive the auto-scaler.
> > +	Thresholds []AutoScaleThreshold `json:"thresholds,omitempty"`
> > +
> > +	// Enabled turns auto scaling on or off.
> > +	Enabled bool `json:"enabled,omitempty"`
> What is the purpose of this field? We don't have "Enabled" for ReplicationController or Pod. I'm not necessarily opposed but I think we should try to be consistent across all the objects.
> 
> —
> Reply to this email directly or view it on GitHub.
>